c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -440,17 +436,21 @@ pub fn visit_body<'tcx>(
         c_void_casts,
     };
 
-    for (bb, bb_data) in mir.basic_blocks().iter_enumerated() {
+    for (block, bb_data) in mir.basic_blocks().iter_enumerated() {
         for (idx, stmt) in bb_data.statements.iter().enumerate() {
-            tc.current_location = Location {
-                block: bb,
+            let loc = Location {
+                block,
                 statement_index: idx,
             };
-            tc.visit_statement(stmt);
+            tc.current_location = loc;
+
+            if !c_void_casts.should_skip_stmt(&loc) {
```

Comment:
I did this to avoid passing `loc` into `visit_statement`, but if it's inappropriate to check `c_void_casts` here I can move it into `visit_statement`